masq_lib/src/percentage.rs

```diff
@@ -80,178 +229,418 @@ impl Percentage {
         // an overflow, so for such large numbers like this one we switch the order of operations.
-        // We're gonna devide it by 100 first and multiple after it. (However, we'd lose some
+        // We're going to divide it by 100 first and multiple after it. (However, we'd lose some
         // precision in smaller numbers that same way). Why that much effort? I don't want to see
-        // an overflow happen where most people would't anticipate it: when going for a percentage
-        // from their number, implaying a request to receive another number, but always smaller
-        // than that passed in.
-        let result = Percentage::new(61).of(quite_large_value);
+        // an overflow happen where most people wouldn't anticipate it: when going for
+        // a PurePercentage from their number, implying a request to receive another number, but
+        // always smaller than that passed in.
+        let case_one = PurePercentage::try_from(61).unwrap().of(u64::MAX / 60);
+        // There is more going on under the hood, which shows better on the following example:
+        // if we divide 255 by 100, we get 2. Then multiplied by 30, it amounts to 60. The right
+        // result, though, is 77 (with an extra 1 from rounding). Therefor there is another
+        // piece of code whose charge is to treat the remainder of modulo 100 that is pushed off
+        // the scoped, and if ignored, it would cause the result to be undervalued. This remainder
+        // is again treated the by the primary (reversed) methodology with num * percents done
+        // first, followed by the final division, keeping just one hundredth.
+        let case_two = PurePercentage::try_from(30).unwrap().of(u8::MAX);
+        // We apply the rounding even here. That's why we'll see the result drop by one compared to
+        // the previous case. As 254 * 30 is 7620, the two least significant digits come rounded
+        // by 100 as 0 which means 7620 divided by 100 makes 76.
+        let case_three = PurePercentage::try_from(30).unwrap().of(u8::MAX - 1);
+
+        assert_eq!(case_one, 187541898082713775);
+        assert_eq!(case_two, 77);
+        assert_eq!(case_three, 76)
+        //Note: Interestingly, this isn't a threat on the negative numbers, even the extremes.
+    }
+
+    #[test]
+    fn zeroes_for_loose_percentage() {
+        assert_eq!(LoosePercentage::new(45).of(0).unwrap(), 0);
+        assert_eq!(LoosePercentage::new(0).of(33).unwrap(), 0)
+    }
+
+    #[test]
+    fn loose_percentage_end_to_end_test_for_standard_values_unsigned() {
+        let expected_values = (0..=100).collect::<Vec<i8>>();
 
-        let expected_result = (quite_large_value / 100) * 61;
-        assert_eq!(result, expected_result);
+        test_end_to_end(100, expected_values, |percent, base| {
+            LoosePercentage::new(percent as u32).of(base).unwrap()
+        })
+    }
```

Comment:
It would be good if you may reorder items in this test similar to an earlier comment, such that base value is on the top, then act and expected value.

You may like to do the same for the test below.